### ISSUE TITLE
FxR becomes unresponsive after leaving immersive mode on sketchfab.com

### ISSUE THREAD

**Issue by bluemarvin** [MozillaReality/FirefoxReality#1163]
## Hardware

Oculus (with and without e10s
## Steps to Reproduce

1. Goto `sketchfab.com`
2. scroll down to the embedded robot 3d model
3. press the play button
4. enter immersive mode.
5. wait for immersive mode to start.
6. exit immersive mode by pressing app button.

## Current Behavior

The navbar is unresponsive. Pressing the back button has no effect (even though we still get the `onLoadRequest()` callback from GeckoView. Page can still be scrolled and javascript animations are still running on the page. Entering new URL in to nav bar doesn't work either. After a long while, FxR eventually become responsive again.

## Expected Behavior

Page should be responsive after exiting immersive mode.

**Comment by bluemarvin** [comment]
This is looking like javascript badness. Still investigating.

**Comment by philip-lamb** [comment]
Performance fixes will land in 1.3 at earliest.

**Comment by philip-lamb** [comment]
I was able to reproduce this with FxR v1.2.3 on Oculus Go. It seems to be some javascript execution problem where it's hogging the CPU and resulting in other threads being starved, including the compositor. If we can nail down some more accurate STR we should probably file this as a GV bug for further investigation

**Comment by keianhzo** [comment]
@philip-lamb @bluemarvin I can still reproduce this with 70.0.20190710094220. At least i was able to reproduce it once because it's so slow in the Go that is almost unplayable.

**Comment by MortimerGoro** [comment]
I can still reproduce it on the Go too. It works much better on the Quest.

**Comment by philip-lamb** [comment]
This may be the same issue raised by Oculus QA #1387

**Comment by bluemarvin** [comment]
This is a different issue. It looks like JavaScript is blocking the main thread.

**Comment by philip-lamb** [comment]
@keianhzo raised a bug where JS code that causes immersive mode to be exited may leave the VR context active. https://bugzilla.mozilla.org/show_bug.cgi?id=1565268 Could this be a possibility here too?

**Comment by jvonitter** [comment]
@bluemarvin can you escalate with the Gecko crew to see what this is?

**Comment by keianhzo** [comment]
@philip-lamb still happening in da676ed29a35570008eab3a369053e2833efeab5

**Comment by bluemarvin** [comment]
sketchfab has become even worse even in the oculus browser. They do not appear to be testing on mobile devices.

**Comment by daoshengmu** [comment]
This might be the root cause, https://bugzilla.mozilla.org/show_bug.cgi?id=1588177.

**Comment by MortimerGoro** [comment]
I've been doing some research today. I found a faster URL to reproduce the issue: https://sketchfab.com/models/9688985db84d447580d40e40e1649407/embed?autostart=0

- Found that the issue doesn't happen if you pause the model animation or set the static pose in the player.
- I added some logs in gecko to check the RAFs and the trigerred window.events and didn't find any infinite loop.
- The issue can be reproduced in multiprocess enabled and disabled. It's easier to reproduce with multiprocess disabled.
- I tried to do some profiler capture but sometimes it's impossible to stop the profiler record and when the times when the record stop works the generated profile.json seems to contain empty data after the freeze starts.

> This might be the root cause, https://bugzilla.mozilla.org/show_bug.cgi?id=1588177.
I tried to revert the mentioned regression in a local GV compilation and it didn't fix the error.
